crates/sortition/src/sortition.rs

```diff
@@ -13,110 +15,328 @@ use e3_events::{
     BusError, CiphernodeAdded, CiphernodeRemoved, EnclaveErrorType, EnclaveEvent, EventBus, Seed,
     Subscribe,
 };
+use serde::{Deserialize, Serialize};
 use std::collections::{HashMap, HashSet};
 use tracing::{info, instrument, trace};
 
+/// Message: ask the `Sortition` actor whether `address` would be in the
+/// committee of size `size` for randomness `seed` on `chain_id`.
+///
+/// Membership semantics depend on the backend for that chain:
+/// - **Distance backend**: computes a committee using address distance.
+/// - **Score backend**: computes each node’s best ticket score and sorts globally.
+///
+/// Returns `true` if `address` appears in the resulting top-`size` selection.
 #[derive(Message, Clone, Debug, PartialEq, Eq)]
 #[rtype(result = "bool")]
 pub struct GetHasNode {
+    /// Round seed / randomness used by the sortition algorithm.
     pub seed: Seed,
+    /// Hex-encoded node address (e.g., `"0x..."`).
     pub address: String,
+    /// Committee size (top-N).
     pub size: usize,
+    /// Target chain.
     pub chain_id: u64,
 }
 
+/// Message: request the current set of registered node addresses for `chain_id`.
+#[derive(Message, Clone, Debug)]
+#[rtype(result = "Vec<String>")]
+pub struct GetNodes {
+    /// Target chain.
+    pub chain_id: u64,
+}
+
+/// Minimal interface that all sortition backends must implement.
+///
+/// Backends can store their own shapes (e.g., a `HashSet<String>` of addresses
+/// for Distance, or a `Vec<RegisteredNode>` for Score), but they must be able to:
+/// - Check committee membership (`contains`)
+/// - Add and remove nodes
+/// - List all registered node addresses
 pub trait SortitionList<T> {
+    /// Return `true` if `address` appears in the size-`size` committee under `seed`.
+    ///
+    /// Implementations should return `Ok(false)` if the backend has no nodes
+    /// or if `size == 0`.
     fn contains(&self, seed: Seed, size: usize, address: T) -> Result<bool>;
```

Comment:
Not a suggestion for this PR but a thought: A short coming of this trait is that we may want to consider changing this to a predicate clsure in the future if we consider `SortitionList<RegisteredNode>`

```rust
if list.contains(seed, size, |n:&RegisteredNode| n.address == address)? {
  //..
}

// also 

let Some(party_id) = list.get_index(seed, size, |n:&RegisteredNode| n.address == address);
```

That might mean that we could add a `put` function to overwrite `RegisteredNode`s and then we could ditch the highly specific `set_ticket_count_addr()` and would be left wit a pretty simple API.